contrib/win32/win32compat/shell-host.c

```diff
@@ -1192,19 +1195,44 @@ ProcessMessages(void* p)
 }
 
 wchar_t *
-w32_cmd_path()
+get_default_shell_path()
 {
+	HKEY reg_key = 0;
+	int tmp_len = PATH_MAX;
 	errno_t r = 0;
-	if ((r = wcsncpy_s(cmd_exe_path, _countof(cmd_exe_path), system32_path, wcsnlen(system32_path, _countof(system32_path)) + 1)) != 0) {
-		printf_s("wcsncpy_s failed with error: %d.", r);
-		exit(255);
+	REGSAM mask = STANDARD_RIGHTS_READ | KEY_QUERY_VALUE | KEY_WOW64_64KEY;
+	wchar_t *tmp = malloc(PATH_MAX + 1);
+
+	memset(tmp, 0, PATH_MAX + 1);
+	memset(default_shell_path, 0, _countof(default_shell_path));
+
+	if ((RegOpenKeyExW(HKEY_LOCAL_MACHINE, L"SOFTWARE\\OpenSSH", 0, mask, &reg_key) == ERROR_SUCCESS) &&
+		(RegQueryValueExW(reg_key, L"DefaultShell", 0, NULL, (LPBYTE)tmp, &tmp_len) == ERROR_SUCCESS) &&
+		(tmp)) {
+		is_default_shell_configured = TRUE;
+		if (wcsstr(default_shell_path, L"bash.exe"))
```

Comment:
??? default_shell_path is being built below. how could it ever be set to bash.exe here?